src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -151,6 +163,8 @@ struct xocl_xgq_vmr {
 	struct semaphore 	xgq_data_sema;
 	struct semaphore 	xgq_log_page_sema;
 	struct xgq_cmd_cq_default_payload xgq_cq_payload;
+	//clock scaling structure to save all feature details
+	struct xgq_clk_scaling_info xgq_cs_struct;
```

Comment:
We should not have this specific additional struct iin xocl_xgq_vmr.
We should have a scaling_payload added in xgq_cmd_cq in the xgq_cmd_vmr.h.
Example code:
struct xgq_cmd_cq {
           hdr;
           union {
                cq_default_payload;
                cq_log_payload; 
                cq_cls_payload; //clock scaling payload? you can have your preferred name
            }